apps/accounts/models/provider_request.py

```diff
@@ -282,26 +286,51 @@ def approve(self) -> Hostingprovider:
                 hostingprovider=hp,
             )
 
+        # Fetch our archived documents:
```

Comment:
This represents a change in our logic.

Because submitted evidence in a verification request and supporting evidence are different kinds of objects, with different IDs, even if they contain the same content, we need a way to see they are "functionally equivalent", to determine if they should be retained on the hosting provider object after an approval.